sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs

```diff
@@ -1249,6 +1250,150 @@ private void SetNameFieldsIfNull()
+        /// Defaults to the beginning of the blob.
+        /// </param>
+        /// <param name="bufferSize">
+        /// The buffer size to use when the stream downloads parts
+        /// of the blob.  Defaults to 1 MB.
+        /// </param>
+        /// <param name="conditions">
+        /// Optional <see cref="BlobRequestConditions"/> to add conditions on
+        /// the download of the blob.
+        /// </param>
+        /// <returns>
+        /// Returns a stream that will download the blob as the stream
+        /// is read from.
+        /// </returns>
+#pragma warning disable AZC0002 // DO ensure all service methods, both asynchronous and synchronous, take an optional CancellationToken parameter called cancellationToken.
+#pragma warning disable AZC0015 // Unexpected client method return type.
+        public virtual Stream OpenRead(
+#pragma warning restore AZC0015 // Unexpected client method return type.
+#pragma warning restore AZC0002 // DO ensure all service methods, both asynchronous and synchronous, take an optional CancellationToken parameter called cancellationToken.
+            long position = 0,
+            int bufferSize = Constants.DefaultStreamingDownloadSize,
+            BlobRequestConditions conditions = default)
+            => OpenReadInteral(
+                position,
+                bufferSize,
+                conditions);
+
+        /// <summary>
+        /// Opens a stream for reading from the blob.  The stream will only download
+        /// the blob as the stream is read from.
+        /// </summary>
+        /// <param name="position">
+        /// The position within the blob to begin the stream.
+        /// Defaults to the beginning of the blob.
+        /// </param>
+        /// <param name="bufferSize">
+        /// The buffer size to use when the stream downloads parts
+        /// of the blob.  Defaults to 1 MB.
+        /// </param>
+        /// <param name="conditions">
+        /// Optional <see cref="BlobRequestConditions"/> to add conditions on
+        /// the download of the blob.
+        /// </param>
+        /// <returns>
+        /// Returns a stream that will download the blob as the stream
+        /// is read from.
+        /// </returns>
+#pragma warning disable AZC0002 // DO ensure all service methods, both asynchronous and synchronous, take an optional CancellationToken parameter called cancellationToken.
+#pragma warning disable AZC0015 // Unexpected client method return type.
+        public virtual Task<Stream> OpenReadAsync(
+#pragma warning restore AZC0015 // Unexpected client method return type.
+#pragma warning restore AZC0002 // DO ensure all service methods, both asynchronous and synchronous, take an optional CancellationToken parameter called cancellationToken.
+            long position = 0,
+            int bufferSize = Constants.DefaultStreamingDownloadSize,
+            BlobRequestConditions conditions = default)
+            => Task.FromResult(OpenReadInteral(
```

Comment:
Use `EnsureCompleted` instead.

Reply:
`OpenReadInternal()` returns a stream, not a Task.

Reply:
Fixed in next iteration.